src/main/java/moze_intel/projecte/integration/jei/world_transmute/WorldTransmuteEntry.java

```diff
@@ -0,0 +1,158 @@
+package moze_intel.projecte.integration.jei.world_transmute;
+
+import mezz.jei.api.ingredients.IIngredients;
+import mezz.jei.api.recipe.IRecipeWrapper;
+import moze_intel.projecte.utils.WorldTransmutations;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockStaticLiquid;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.init.Blocks;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidRegistry;
+import net.minecraftforge.fluids.FluidStack;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class WorldTransmuteEntry implements IRecipeWrapper
+{
+	private ItemStack inputItem = ItemStack.EMPTY;
+	private ItemStack leftOutputItem = ItemStack.EMPTY;
+	private ItemStack rightOutputItem = ItemStack.EMPTY;
+	private FluidStack inputFluid;
+	private FluidStack leftOutputFluid;
+	private FluidStack rightOutputFluid;
+
+	public WorldTransmuteEntry(WorldTransmutations.Entry transmutationEntry)
+	{
+		Block inputBlock = transmutationEntry.input.getBlock();
+		IBlockState leftOutput = transmutationEntry.outputs.getLeft();
+		IBlockState rightOutput = transmutationEntry.outputs.getRight();
+
+		inputFluid = fluidFromBlock(inputBlock);
+		if (inputFluid == null)
+		{
+			inputItem = itemFromBlock(inputBlock, transmutationEntry.input);
+		}
+		if (leftOutput != null)
+		{
+			leftOutputFluid = fluidFromBlock(leftOutput.getBlock());
+			if (leftOutputFluid == null)
+			{
+				leftOutputItem = itemFromBlock(leftOutput.getBlock(), leftOutput);
+			}
+		}
+		if (rightOutput != null)
+		{
+			rightOutputFluid = fluidFromBlock(rightOutput.getBlock());
+			if (rightOutputFluid == null)
+			{
+				rightOutputItem = itemFromBlock(rightOutput.getBlock(), rightOutput);
+			}
+		}
+	}
+
+	private FluidStack fluidFromBlock(Block block)
+	{
+		if (block instanceof BlockStaticLiquid)
```

Comment:
what's the point if we just do == below anyway?

Reply:
There isn't I will remove it. It was from the current handling except I have the other case elsewhere so it makes the added if statement not needed at all.